### ISSUE TITLE
[BUG] Error while saving Node Disk setting when there is no tag

### ISSUE THREAD

**Issue by kaxing** [longhorn/longhorn#3538]
## Describe the bug

![image](https://user-images.githubusercontent.com/602540/149709989-ec8761f8-6781-43cb-af84-5f8e9e46473e.png)

## To Reproduce

1. Go to Nodes page
2. Click on Edit (any node)
3. Click save

## Expected behavior

Node or Disk setting should work without fill in tags.

## Log or Support bundle

[longhorn-support-bundle_2e895d91-95a1-41c7-8ec1-a8abddb10dab_2022-01-17T04-49-51Z.zip](https://github.com/longhorn/longhorn/files/7879059/longhorn-support-bundle_2e895d91-95a1-41c7-8ec1-a8abddb10dab_2022-01-17T04-49-51Z.zip)

## Environment

 - Longhorn version: master-head (longhorn-manager repo)
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): kubectl apply
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: k3s 1.18
   - Number of management node in the cluster: 1
   - Number of worker node in the cluster: 3
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): HyperKit VM

## Additional context

```
[longhorn-manager-k54x6 longhorn-manager] time="2022-01-17T04:52:30Z" level=warning msg="HTTP handling error Node.longhorn.io \"c1-worker1\" is invalid: spec.tags: Invalid value: \"null\": spec.tags in body must be of type array: \"null\""
[longhorn-manager-k54x6 longhorn-manager] time="2022-01-17T04:52:30Z" level=error msg="Error in request: Node.longhorn.io \"c1-worker1\" is invalid: spec.tags: Invalid value: \"null\": spec.tags in body must be of type array: \"null\""
[longhorn-manager-k54x6 longhorn-manager] 10.42.1.1 - - [17/Jan/2022:04:52:30 +0000] "PUT /v1/nodes/c1-worker1 HTTP/1.1" 500 285 "http://192.168.64.20:30419/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36 Edg/95.0.1020.53"
[longhorn-manager-k54x6 longhorn-manager] time="2022-01-17T04:52:30Z" level=warning msg="HTTP handling error Node.longhorn.io \"c1-worker1\" is invalid: spec.disks.default-disk-e22d292bdb653860.tags: Invalid value: \"null\": spec.disks.default-disk-e22d292bdb653860.tags in body must be of type array: \"null\""
[longhorn-manager-k54x6 longhorn-manager] time="2022-01-17T04:52:30Z" level=error msg="Error in request: Node.longhorn.io \"c1-worker1\" is invalid: spec.disks.default-disk-e22d292bdb653860.tags: Invalid value: \"null\": spec.disks.default-disk-e22d292bdb653860.tags in body must be of type array: \"null\""
[longhorn-manager-k54x6 longhorn-manager] 10.42.1.1 - - [17/Jan/2022:04:52:30 +0000] "POST /v1/nodes/c1-worker1?action=diskUpdate HTTP/1.1" 500 357 "http://192.168.64.20:30419/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.69 Safari/537.36 Edg/95.0.1020.53"
[longhorn-manager-k54x6 longhorn-manager] time="2022-01-17T04:52:30Z" level=debug msg="Requeue default due to conflict" controller=longhorn-backup-target cred= error="Operation cannot be fulfilled on backuptargets.longhorn.io \"default\": StorageError: invalid object, Code: 4, Key: /registry/longhorn.io/backuptargets/longhorn-system/default, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: 95d5ee7e-4bea-47f0-b434-7a8bd5b286ea, UID in object meta: " interval=5m0s node=c1-worker2 url=
```

**Comment by innobead** [comment]
@derekbit This should be about backward compatibility after introducing the new CRD version.

**Comment by derekbit** [comment]
`node.spec.disks.tags=nil` cannot pass the validation after introducing structural CRD schema, so need to initialize the tags.

**Comment by innobead** [comment]
> `node.spec.disks.tags=nil` cannot pass the validation after introducing structural CRD schema, so need to initialize the tags if it is null.

Maybe just add `// +nullable` for that property. We probably need to check again to see what properties are nullable.

**Comment by kaxing** [comment]
Editing the nodes.longhorn.io cr directly won't giving error with invalid tag, however it can be display on UI without problem, even can be deleted.
```yaml
  tags:
  - tag2
  - tag4
  - --tag4
  - tag4--
```
![image](https://user-images.githubusercontent.com/602540/150951783-a0b589d1-5e29-4007-8f33-bf6fe9809e31.png)

I think it will be good to have schema for `nodes.longhorn.io `. cc @derekbit @jenting

**Comment by kaxing** [comment]
Under testing with master-head,

Editing the nodes.longhorn.io cr directly won't giving error with invalid tag, however it can be display on UI without problem, even can be deleted.
```yaml
  tags:
  - tag2
  - tag4
  - --tag4
  - tag4--
```
![image](https://user-images.githubusercontent.com/602540/150951783-a0b589d1-5e29-4007-8f33-bf6fe9809e31.png)

I think it will be good to have schema for `nodes.longhorn.io `. cc @derekbit @jenting

**Comment by kaxing** [comment]
Validation - **PASSED**
Tested on master-head.
No more error while saving node info with empty tag list from the UI.

The _kubectl edit nodes.longhorn.io/node-name_ validation will be implemented in #3562

**Comment by derekbit** [comment]
> Under testing with master-head,
> 
> Editing the nodes.longhorn.io cr directly won't giving error with invalid tag, however it can be display on UI without problem, even can be deleted.
> 
> ```yaml
>   tags:
>   - tag2
>   - tag4
>   - --tag4
>   - tag4--
> ```
> 
> ![image](https://user-images.githubusercontent.com/602540/150951783-a0b589d1-5e29-4007-8f33-bf6fe9809e31.png)
> 
> I think it will be good to have schema for `nodes.longhorn.io `. cc @derekbit @jenting

This will be validated in the validating webhook.
Related to https://github.com/longhorn/longhorn/issues/3562
